app/src/main/java/com/sun/ui/base/BasePagingAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.sun.ui.base
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import com.sun.BR
+import com.sun.data.model.BaseModelPaging
+import com.sun.ui.sample.activity.RepoListener
+
+class BasePagingAdapter<T: Any>(
+    @LayoutRes private val layoutRes: Int,
+    private val listener: RepoListener,
+    comparator: DiffUtil.ItemCallback<T>
+): PagingDataAdapter<T, BaseViewHolder<ViewDataBinding>>(comparator) {
```

Comment:
truyền luôn comparator vào luôn như này đc k bạn?
```suggestion
class BasePagingAdapter<T: BaseModelPaging>(
    @LayoutRes private val layoutRes: Int,
    private val listener: RepoListener
): PagingDataAdapter<T, BaseViewHolder<ViewDataBinding>>(BaseComparator<T>) {
````

Reply:
hồi t có thử viết để code gọn nhất có thể mà không được